### PR TITLE
Add goodreads as provider

### PR DESCRIPTION
**Pull request by mstruebing**

### CODE REVIEW

**Comment by mstruebing** [comment]
I don't know why the build is failing, maybe a restart of the failing jobs could help?

File: providers/goodreads/goodreads.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+type Provider struct{}
+
+// BuildURI generates a search URL for Goodreads.
+func (p *Provider) BuildURI(q string) string {
+	return fmt.Sprintf("https://www.goodreads.com/search?q=%s", url.QueryEscape(q))
+}
+
+// Tags returns the tags relevant to this provider.
+func (p *Provider) Tags() []string {
+	return []string{"search"}
```
Comment (zquestz) [inline providers/goodreads/goodreads.go]: I would remove this tag, this isn't a general purpose search provider.

Comment (mstruebing) [reply providers/goodreads/goodreads.go]: @zquestz fixed.

**Comment by zquestz** [comment]
Thanks for the update!

**Pull request merged**